### PR TITLE
PXC-3340: Merge codership/galera 25.3.30 -> percona/galera 3.x

### PR DESCRIPTION
**Pull request by kamil-holubicki**
Merge codership/galera tag 'release_25.3.30'

### CODE REVIEW

File: galera/src/replicator_str.cpp [hunk 1]
```diff
@@ -46,32 +45,8 @@ ReplicatorSMM::state_transfer_required(const wsrep_view_info_t& view_info)
                     abort();
                 }
 
-                return (local_seqno != group_seqno);
+                return (local_seqno < group_seqno);
```
Comment (dutow) [inline galera/src/replicator_str.cpp]: Additional whitespace at the end.

File: galera/src/replicator_str.cpp [hunk 2]
```diff
@@ -820,19 +795,12 @@ ReplicatorSMM::request_state_transfer (void* recv_ctx,
...
-    send_state_request (req);
-=======
-    GU_DBUG_SYNC_WAIT("before_send_state_request");
-    send_state_request (req);
->>>>>>> release_25.3.30
-
     sst_state_ = SST_WAIT;
-<<<<<<< HEAD
+
+    GU_DBUG_SYNC_WAIT("before_send_state_request");
```
Comment (dutow) [inline galera/src/replicator_str.cpp]: Shouldn't this go before the sst_state_ change / comment? Based on the comment those seem to be related, and then this should go before both.

File: gcache/src/gcache_rb_store.cpp [hunk 3]
```diff
@@ -1035,10 +1035,6 @@ namespace gcache
             else assert(size_trail_ >= sizeof(BufferHeader));
 
             estimate_space();
-            /* On graceful shutdown all the active buffers are released
```
Comment (dutow) [inline gcache/src/gcache_rb_store.cpp:1038]: This isn't a conflict resolution? Please explain/mention in the commit message.

File: galera/src/replicator_str.cpp (hunk 1 above)
Comment (kamil-holubicki) [reply galera/src/replicator_str.cpp]: Fixed

File: galera/src/replicator_str.cpp (hunk 2 above)
Comment (kamil-holubicki) [reply galera/src/replicator_str.cpp]: fixed

File: gcache/src/gcache_rb_store.cpp (hunk 3 above)
Comment (kamil-holubicki) [reply gcache/src/gcache_rb_store.cpp:1038]: Fixed

**Review by dutow — APPROVED** [review]

**Pull request merged**